cloudcompose/cluster/aws/cloudcontroller.py

```diff
@@ -262,6 +266,16 @@ def _create_instances(self, block_device_map, cloud_init):
                 prefix = 'created'
             print "%s %s %s (%s)" % (prefix, instance_id, instance_name, private_ip)
 
+    def _cloud_init_build(self, cloud_init, **kwargs):
+        cloud_init_script = cloud_init.build(self.config_data, **kwargs)
+        if len(cloud_init_script) > MAX_CLOUD_INIT_LENGTH:
+            output = StringIO()
+            with GzipFile(mode='wb', fileobj=output) as gzfile:
+                gzfile.write(cloud_init_script)
+            cloud_init_script = "#!/bin/sh\necho '%s' | base64 -d | gunzip | sh" % b64encode(output.getvalue())
```

Comment:
It will throw an error saying from the AWS API saying it is too large. I would rather do that so if they raise the limit I don't have to raise it in my code.